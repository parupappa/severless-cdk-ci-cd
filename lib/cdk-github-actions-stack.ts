import * as cdk from '@aws-cdk/core';
import * as lambda from '@aws-cdk/aws-lambda';
import * as apigateway from '@aws-cdk/aws-apigateway';
export class CdkGithubActionsStack extends cdk.Stack {
  constructor(scope: cdk.App, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    // The code that defines your stack goes here
    
    // Create a Lambda function with code from the "src/lambda" directory
    const hello = new lambda.Function(this, 'HelloHandler', {
      runtime: lambda.Runtime.NODEJS_16_X,
      code: lambda.Code.fromAsset('src/'),
      handler: 'hello.handler',
    });

    // defines an API Gateway REST API resource backed by our "hello" function.
    new apigateway.LambdaRestApi(this, 'GithubActionsEndpoint', {
      handler: hello,
      endpointTypes: [ apigateway.EndpointType.REGIONAL ],
    });
  }
}
