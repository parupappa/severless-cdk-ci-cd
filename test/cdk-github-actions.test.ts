import { App } from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import { CdkGithubActionsStack } from '../lib/cdk-github-actions-stack';

test('SnapshotTest', () => {
  const app = new App();
  const stack = new CdkGithubActionsStack(app, 'CdkGithubActionsStack', {});
  const template = Template.fromStack(stack);
  expect(template.toJSON()).toMatchSnapshot();
});