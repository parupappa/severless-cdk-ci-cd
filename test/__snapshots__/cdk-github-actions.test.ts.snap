// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SnapshotTest 1`] = `
{
  "Outputs": {
    "GithubActionsEndpoint75B41E05": {
      "Value": {
        "Fn::Join": [
          "",
          [
            "https://",
            {
              "Ref": "GithubActionsEndpoint4808611A",
            },
            ".execute-api.",
            {
              "Ref": "AWS::Region",
            },
            ".",
            {
              "Ref": "AWS::URLSuffix",
            },
            "/",
            {
              "Ref": "GithubActionsEndpointDeploymentStageprod38921B69",
            },
            "/",
          ],
        ],
      },
    },
  },
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "GithubActionsEndpoint4808611A": {
      "Properties": {
        "EndpointConfiguration": {
          "Types": [
            "REGIONAL",
          ],
        },
        "Name": "GithubActionsEndpoint",
      },
      "Type": "AWS::ApiGateway::RestApi",
    },
    "GithubActionsEndpointANYApiPermissionCdkGithubActionsStackGithubActionsEndpoint3647534FANYB93CE56D": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "HelloHandler2E4FBA4D",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              {
                "Ref": "AWS::Region",
              },
              ":",
              {
                "Ref": "AWS::AccountId",
              },
              ":",
              {
                "Ref": "GithubActionsEndpoint4808611A",
              },
              "/",
              {
                "Ref": "GithubActionsEndpointDeploymentStageprod38921B69",
              },
              "/*/",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "GithubActionsEndpointANYApiPermissionTestCdkGithubActionsStackGithubActionsEndpoint3647534FANY37B035BE": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "HelloHandler2E4FBA4D",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              {
                "Ref": "AWS::Region",
              },
              ":",
              {
                "Ref": "AWS::AccountId",
              },
              ":",
              {
                "Ref": "GithubActionsEndpoint4808611A",
              },
              "/test-invoke-stage/*/",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "GithubActionsEndpointANYE0AB0F88": {
      "Properties": {
        "AuthorizationType": "NONE",
        "HttpMethod": "ANY",
        "Integration": {
          "IntegrationHttpMethod": "POST",
          "Type": "AWS_PROXY",
          "Uri": {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":apigateway:",
                {
                  "Ref": "AWS::Region",
                },
                ":lambda:path/2015-03-31/functions/",
                {
                  "Fn::GetAtt": [
                    "HelloHandler2E4FBA4D",
                    "Arn",
                  ],
                },
                "/invocations",
              ],
            ],
          },
        },
        "ResourceId": {
          "Fn::GetAtt": [
            "GithubActionsEndpoint4808611A",
            "RootResourceId",
          ],
        },
        "RestApiId": {
          "Ref": "GithubActionsEndpoint4808611A",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
    "GithubActionsEndpointAccount09F4903E": {
      "DeletionPolicy": "Retain",
      "DependsOn": [
        "GithubActionsEndpoint4808611A",
      ],
      "Properties": {
        "CloudWatchRoleArn": {
          "Fn::GetAtt": [
            "GithubActionsEndpointCloudWatchRole8D18CAD1",
            "Arn",
          ],
        },
      },
      "Type": "AWS::ApiGateway::Account",
      "UpdateReplacePolicy": "Retain",
    },
    "GithubActionsEndpointCloudWatchRole8D18CAD1": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "apigateway.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AmazonAPIGatewayPushToCloudWatchLogs",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
      "UpdateReplacePolicy": "Retain",
    },
    "GithubActionsEndpointDeployment1FE27B85de0dfbfeb15ff7aa82cbae22194eaa67": {
      "DependsOn": [
        "GithubActionsEndpointproxyANYE2FFAEBF",
        "GithubActionsEndpointproxyB6804B8F",
        "GithubActionsEndpointANYE0AB0F88",
      ],
      "Properties": {
        "Description": "Automatically created by the RestApi construct",
        "RestApiId": {
          "Ref": "GithubActionsEndpoint4808611A",
        },
      },
      "Type": "AWS::ApiGateway::Deployment",
    },
    "GithubActionsEndpointDeploymentStageprod38921B69": {
      "DependsOn": [
        "GithubActionsEndpointAccount09F4903E",
      ],
      "Properties": {
        "DeploymentId": {
          "Ref": "GithubActionsEndpointDeployment1FE27B85de0dfbfeb15ff7aa82cbae22194eaa67",
        },
        "RestApiId": {
          "Ref": "GithubActionsEndpoint4808611A",
        },
        "StageName": "prod",
      },
      "Type": "AWS::ApiGateway::Stage",
    },
    "GithubActionsEndpointproxyANYApiPermissionCdkGithubActionsStackGithubActionsEndpoint3647534FANYproxy9EDFB17E": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "HelloHandler2E4FBA4D",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              {
                "Ref": "AWS::Region",
              },
              ":",
              {
                "Ref": "AWS::AccountId",
              },
              ":",
              {
                "Ref": "GithubActionsEndpoint4808611A",
              },
              "/",
              {
                "Ref": "GithubActionsEndpointDeploymentStageprod38921B69",
              },
              "/*/*",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "GithubActionsEndpointproxyANYApiPermissionTestCdkGithubActionsStackGithubActionsEndpoint3647534FANYproxy6441872D": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "HelloHandler2E4FBA4D",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              {
                "Ref": "AWS::Region",
              },
              ":",
              {
                "Ref": "AWS::AccountId",
              },
              ":",
              {
                "Ref": "GithubActionsEndpoint4808611A",
              },
              "/test-invoke-stage/*/*",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "GithubActionsEndpointproxyANYE2FFAEBF": {
      "Properties": {
        "AuthorizationType": "NONE",
        "HttpMethod": "ANY",
        "Integration": {
          "IntegrationHttpMethod": "POST",
          "Type": "AWS_PROXY",
          "Uri": {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":apigateway:",
                {
                  "Ref": "AWS::Region",
                },
                ":lambda:path/2015-03-31/functions/",
                {
                  "Fn::GetAtt": [
                    "HelloHandler2E4FBA4D",
                    "Arn",
                  ],
                },
                "/invocations",
              ],
            ],
          },
        },
        "ResourceId": {
          "Ref": "GithubActionsEndpointproxyB6804B8F",
        },
        "RestApiId": {
          "Ref": "GithubActionsEndpoint4808611A",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
    "GithubActionsEndpointproxyB6804B8F": {
      "Properties": {
        "ParentId": {
          "Fn::GetAtt": [
            "GithubActionsEndpoint4808611A",
            "RootResourceId",
          ],
        },
        "PathPart": "{proxy+}",
        "RestApiId": {
          "Ref": "GithubActionsEndpoint4808611A",
        },
      },
      "Type": "AWS::ApiGateway::Resource",
    },
    "HelloHandler2E4FBA4D": {
      "DependsOn": [
        "HelloHandlerServiceRole11EF7C63",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "ce09bf490209138df7c9fe49ed0feff5f3a46856d04ebe3d44b70182c4986fdc.zip",
        },
        "Handler": "hello.handler",
        "Role": {
          "Fn::GetAtt": [
            "HelloHandlerServiceRole11EF7C63",
            "Arn",
          ],
        },
        "Runtime": "nodejs16.x",
      },
      "Type": "AWS::Lambda::Function",
    },
    "HelloHandlerServiceRole11EF7C63": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;

exports[`testStack 1`] = `
{
  "Outputs": {
    "GithubActionsEndpoint75B41E05": {
      "Value": {
        "Fn::Join": [
          "",
          [
            "https://",
            {
              "Ref": "GithubActionsEndpoint4808611A",
            },
            ".execute-api.",
            {
              "Ref": "AWS::Region",
            },
            ".",
            {
              "Ref": "AWS::URLSuffix",
            },
            "/",
            {
              "Ref": "GithubActionsEndpointDeploymentStageprod38921B69",
            },
            "/",
          ],
        ],
      },
    },
  },
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "GithubActionsEndpoint4808611A": {
      "Properties": {
        "EndpointConfiguration": {
          "Types": [
            "REGIONAL",
          ],
        },
        "Name": "GithubActionsEndpoint",
      },
      "Type": "AWS::ApiGateway::RestApi",
    },
    "GithubActionsEndpointANYApiPermissionCdkGithubActionsStackGithubActionsEndpoint3647534FANYB93CE56D": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "HelloHandler2E4FBA4D",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              {
                "Ref": "AWS::Region",
              },
              ":",
              {
                "Ref": "AWS::AccountId",
              },
              ":",
              {
                "Ref": "GithubActionsEndpoint4808611A",
              },
              "/",
              {
                "Ref": "GithubActionsEndpointDeploymentStageprod38921B69",
              },
              "/*/",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "GithubActionsEndpointANYApiPermissionTestCdkGithubActionsStackGithubActionsEndpoint3647534FANY37B035BE": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "HelloHandler2E4FBA4D",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              {
                "Ref": "AWS::Region",
              },
              ":",
              {
                "Ref": "AWS::AccountId",
              },
              ":",
              {
                "Ref": "GithubActionsEndpoint4808611A",
              },
              "/test-invoke-stage/*/",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "GithubActionsEndpointANYE0AB0F88": {
      "Properties": {
        "AuthorizationType": "NONE",
        "HttpMethod": "ANY",
        "Integration": {
          "IntegrationHttpMethod": "POST",
          "Type": "AWS_PROXY",
          "Uri": {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":apigateway:",
                {
                  "Ref": "AWS::Region",
                },
                ":lambda:path/2015-03-31/functions/",
                {
                  "Fn::GetAtt": [
                    "HelloHandler2E4FBA4D",
                    "Arn",
                  ],
                },
                "/invocations",
              ],
            ],
          },
        },
        "ResourceId": {
          "Fn::GetAtt": [
            "GithubActionsEndpoint4808611A",
            "RootResourceId",
          ],
        },
        "RestApiId": {
          "Ref": "GithubActionsEndpoint4808611A",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
    "GithubActionsEndpointAccount09F4903E": {
      "DeletionPolicy": "Retain",
      "DependsOn": [
        "GithubActionsEndpoint4808611A",
      ],
      "Properties": {
        "CloudWatchRoleArn": {
          "Fn::GetAtt": [
            "GithubActionsEndpointCloudWatchRole8D18CAD1",
            "Arn",
          ],
        },
      },
      "Type": "AWS::ApiGateway::Account",
      "UpdateReplacePolicy": "Retain",
    },
    "GithubActionsEndpointCloudWatchRole8D18CAD1": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "apigateway.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AmazonAPIGatewayPushToCloudWatchLogs",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
      "UpdateReplacePolicy": "Retain",
    },
    "GithubActionsEndpointDeployment1FE27B85de0dfbfeb15ff7aa82cbae22194eaa67": {
      "DependsOn": [
        "GithubActionsEndpointproxyANYE2FFAEBF",
        "GithubActionsEndpointproxyB6804B8F",
        "GithubActionsEndpointANYE0AB0F88",
      ],
      "Properties": {
        "Description": "Automatically created by the RestApi construct",
        "RestApiId": {
          "Ref": "GithubActionsEndpoint4808611A",
        },
      },
      "Type": "AWS::ApiGateway::Deployment",
    },
    "GithubActionsEndpointDeploymentStageprod38921B69": {
      "DependsOn": [
        "GithubActionsEndpointAccount09F4903E",
      ],
      "Properties": {
        "DeploymentId": {
          "Ref": "GithubActionsEndpointDeployment1FE27B85de0dfbfeb15ff7aa82cbae22194eaa67",
        },
        "RestApiId": {
          "Ref": "GithubActionsEndpoint4808611A",
        },
        "StageName": "prod",
      },
      "Type": "AWS::ApiGateway::Stage",
    },
    "GithubActionsEndpointproxyANYApiPermissionCdkGithubActionsStackGithubActionsEndpoint3647534FANYproxy9EDFB17E": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "HelloHandler2E4FBA4D",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              {
                "Ref": "AWS::Region",
              },
              ":",
              {
                "Ref": "AWS::AccountId",
              },
              ":",
              {
                "Ref": "GithubActionsEndpoint4808611A",
              },
              "/",
              {
                "Ref": "GithubActionsEndpointDeploymentStageprod38921B69",
              },
              "/*/*",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "GithubActionsEndpointproxyANYApiPermissionTestCdkGithubActionsStackGithubActionsEndpoint3647534FANYproxy6441872D": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "HelloHandler2E4FBA4D",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              {
                "Ref": "AWS::Region",
              },
              ":",
              {
                "Ref": "AWS::AccountId",
              },
              ":",
              {
                "Ref": "GithubActionsEndpoint4808611A",
              },
              "/test-invoke-stage/*/*",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "GithubActionsEndpointproxyANYE2FFAEBF": {
      "Properties": {
        "AuthorizationType": "NONE",
        "HttpMethod": "ANY",
        "Integration": {
          "IntegrationHttpMethod": "POST",
          "Type": "AWS_PROXY",
          "Uri": {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":apigateway:",
                {
                  "Ref": "AWS::Region",
                },
                ":lambda:path/2015-03-31/functions/",
                {
                  "Fn::GetAtt": [
                    "HelloHandler2E4FBA4D",
                    "Arn",
                  ],
                },
                "/invocations",
              ],
            ],
          },
        },
        "ResourceId": {
          "Ref": "GithubActionsEndpointproxyB6804B8F",
        },
        "RestApiId": {
          "Ref": "GithubActionsEndpoint4808611A",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
    "GithubActionsEndpointproxyB6804B8F": {
      "Properties": {
        "ParentId": {
          "Fn::GetAtt": [
            "GithubActionsEndpoint4808611A",
            "RootResourceId",
          ],
        },
        "PathPart": "{proxy+}",
        "RestApiId": {
          "Ref": "GithubActionsEndpoint4808611A",
        },
      },
      "Type": "AWS::ApiGateway::Resource",
    },
    "HelloHandler2E4FBA4D": {
      "DependsOn": [
        "HelloHandlerServiceRole11EF7C63",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "ce09bf490209138df7c9fe49ed0feff5f3a46856d04ebe3d44b70182c4986fdc.zip",
        },
        "Handler": "hello.handler",
        "Role": {
          "Fn::GetAtt": [
            "HelloHandlerServiceRole11EF7C63",
            "Arn",
          ],
        },
        "Runtime": "nodejs16.x",
      },
      "Type": "AWS::Lambda::Function",
    },
    "HelloHandlerServiceRole11EF7C63": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
